client/src/game/ui/HanabiCard.ts

```diff
@@ -909,9 +909,10 @@ export default class HanabiCard extends Konva.Group {
   }
 
   // This card was either played or discarded (or revealed at the end of the game)
-  reveal(msgSuit: number, rank: number) {
+  reveal(msgSuit: number, msgRank: number) {
     // Local variables
-    const suit = msgSuitToSuit(msgSuit, globals.variant);
+    const suit = msgSuit === -1 ? null : msgSuitToSuit(msgSuit, globals.variant);
```

Comment:
I'd fix this in msgSuitToSuit since the same issue happens elsewhere (and remove the check there)
```
export const msgSuitToSuit = (
  msgSuit: number,
  variant: Variant,
) => {
  if (msgSuit < 0 || msgSuit >= variant.suits.length) {
    return null;
  }
  return variant.suits[msgSuit];
}
```

Reply:
The opposite is also true, we're depending on the behavior of indexOf (returning -1 when not found) on suitToMsgSuit, might as well make it explicit.

Reply:
Noted, I was just copying some code from action.ts so I'll fix it there too